test/new-e2e/ndm/snmp/snmpTestEnv.go

```diff
@@ -93,10 +94,10 @@ func NewTestEnv() (*TestEnv, error) {
 		envVars := map[string]string{"DATA_DIR": dataPath, "CONFIG_DIR": configPath}
 		composeDependencies := []pulumi.Resource{createDataDirCommand, configCommand}
 		composeDependencies = append(composeDependencies, fileCommands...)
-		_, err = docker.NewAgentDockerInstaller(
-			vm.UnixVM,
```

Comment:
I think this will not have the expected behaviour. With the new `NewDaemon` function, the VM is created automatically here in the function. It means here that all these [line](https://github.com/DataDog/datadog-agent/pull/18453/files#diff-4b47c437598feca25c1b2dacb0bc9f7c0acd5cf48dcbc333d87610b9a3fb158eL55-R91) will create/copy file on a VM that will not be used by the `NewDaemon` function